### PR TITLE
Fix a crash when opening a new web view window

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14866
affects: student
release note: Various bug fixes

test plan: see ticket

### CODE REVIEW

File: Core/Core/DocViewer/DocViewerViewController.swift [hunk 1]
```diff
@@ -101,7 +101,8 @@ public class DocViewerViewController: UIViewController {
         let document = Document(url: localURL)
         if let annotationMeta = metadata.annotations {
             document.defaultAnnotationUsername = annotationMeta.user_name
-            document.didCreateDocumentProviderBlock = { documentProvider in
+            document.didCreateDocumentProviderBlock = { [weak self] documentProvider in
```
Comment (armstrongnate) [inline Core/Core/DocViewer/DocViewerViewController.swift:104]: And this fixes a memory leak I spotted while trying to track down the original crash.

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53080987/p/a747e0d9c086729084171f468bb1ef6d">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53080992/p/381eef8948944a3f3f76cca4a5064126">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53080975/p/094efe3f2781647b3ab1d2aa178b2c46">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  The total test coverage is below the minimum 90%
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  One or more files are below the minimum test coverage 50%
  </td>
    </tr>
  </tbody>
</table>



Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 89.1% | -- | --
Student/Student/Courses/CourseNavigation/CourseNavigationPresenter.swift | 0% | -- | --
Student/Student/Assignments/AssignmentList/AssignmentListModel.swift | 0% | -- | --
Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift | 0% | -- | --
Student/Student/Extensions/UIViewControllerExtensions.swift | 0% | -- | --
Student/Student/Submissions/SubmissionRubric/RubricPresenter.swift | 0% | -- | --
Student/Student/Submissions/SubmissionDetails/SubmissionDetailsPresenter.swift | 0% | -- | --
Student/Student/Submissions/SubmissionComments/SubmissionCommentsPresenter.swift | 0% | -- | --
Student/Student/Routes.swift | 0% | -- | --
Student/Student/Submissions/urlSubmission/UrlSubmissionPresenter.swift | 0% | -- | --
Student/Student/Submissions/ArcSubmission/ArcSubmissionPresenter.swift | 0% | -- | --

[MBL-14866](https://instructure.atlassian.net/browse/MBL-14866)

#### Affected Apps: Student

#### Release Note: 
Various bug fixes
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Review by Attila-Varga-Instructure — APPROVED** [review]

**Comment by Attila-Varga-Instructure** [comment]
Approved.

File: Core/Core/UIViews/CoreWebView.swift [hunk 2]
```diff
@@ -391,7 +402,8 @@ extension CoreWebView: WKUIDelegate {
         guard let from = linkDelegate?.routeLinksFrom else { return nil }
         let controller = CoreWebViewController()
         configuration.userContentController.removeAllUserScripts()
```
Comment (vanwagonet) [inline Core/Core/UIViews/CoreWebView.swift:404]: if we are not going to re-run the setup, should we keep user scripts intact? We only removed them so it wouldn't crash re-adding them.

**Review by vanwagonet — CHANGES_REQUESTED** [review]

File: Core/Core/UIViews/CoreWebView.swift (hunk 2 above)
Comment (armstrongnate) [reply Core/Core/UIViews/CoreWebView.swift:404]: Yeah I think that makes sense.

**Review by vanwagonet — APPROVED** [review]
QA+1

**Pull request merged**